compiler/luci/service/src/CircleShapeInferenceHelper.cpp

```diff
@@ -161,6 +162,40 @@ loco::TensorShape broadcast_shape(const loco::TensorShape &x, const loco::Tensor
   return output_shape;
 }
 
+loco::TensorShape own_shape(const luci::CircleNode *node)
```

Comment:
I don't think copying this method here is a good idea.

Reply:
> I don't think copying this method here is a good idea.

Thanks for the feedback.
I was migrating shape inference rule from `compiler/luci/service/src/CircleShapeInferenceRule.cpp` to `compiler/luci/service/src/Nodes/CircleReshape.cpp`.
And I found out that `own_shape` is being used in many inference logics including reshape.
So I copied it to `CircleShapeInferenceHelper.cpp`.

I think it could be redundant.
Since it was in a private namespace, I could't use it in `compiler/luci/service/src/Nodes/CircleReshape.cpp`.
Could you give me some advice for how to use `own_shape` in other files?

Reply:
I need to explain about this method.
existing shape inference isn't for dynamic shape, began with only static shape, historically.
but then there appeared models with unknown batch (dim 0 for rank 4 shape).
our codes had to mitigate this situation but anyway our main target needed only static shape,
so we've done some shape information passing from input to output (not intelligent shape inference) or do some minimum shape inference or just reuse the shape information from the file.

this `own_shape()` to just to pass what ever was in the first place when our poor shape inference logic couldn't handle it.

as I looked in to this PR, this PR tells it's doing some smart shape inference (by the comment) and should not use this method if the logic is well read and understood.

anyway, current PR has lots of changes in multiple purposes (as the comment says) that would be better to be separated for single purpose.

FYI @shs-park , @zetwhite

Reply:
Thanks a lot.
I have fully understood about the `own_shape` method.
I'm planning to not to use `own_shape` in my new shape inference algorithm.

And about the current PR, I admit it's very complicated.
Sorry for the confusion. It was my first commitment to an open source project and I had to be more careful.

I think I need to start over my PR from draft with following 3 separate parts:
1. Migrate current reshape algorithm to `sinf::Algorithm`
2. Implement dynamic shape inference algorithm for const inputs.
3. Implement dynamic shape inference algorithm for non-const inputs.

If you don't mind I'll close this PR and start over with my SSAFY mentors :)

Reply:
> If you don't mind I'll close this PR and start over with my SSAFY mentors :)

no problem and it's OK with me.